src/tree/v2/CompatibleBranchDataItem.ts

```diff
@@ -27,20 +27,18 @@ export class CompatibleBranchDataItem extends BranchDataItem {
     public get quickPickOptions(): CompatibleQuickPickOptions {
         const ti = this.branchItem as AzExtTreeItem;
 
-        const maybeParent = ti as AzExtParentTreeItem;
-
-        const createChild: CreateOptions | undefined = maybeParent.createChild ?
+        const createChild: CreateOptions | undefined = isAzExtParentTreeItem(ti) ?
             {
-                callback: maybeParent.createChild.bind(maybeParent) as typeof maybeParent.createChild,
-                label: maybeParent.childTypeLabel ?
-                    localize('createNewItem', '$(plus) Create new {0}', maybeParent.childTypeLabel) :
+                callback: ti.createChild.bind(ti) as typeof ti.createChild,
+                label: ti.childTypeLabel ?
+                    localize('createNewItem', '$(plus) Create new {0}', ti.childTypeLabel) :
                     localize('createNew', '$(plus) Create new...'),
             } :
             undefined;
 
         return {
             contextValues: ti.contextValue.split(';'),
-            isLeaf: !maybeParent?.loadMoreChildrenImpl,
+            isLeaf: isAzExtParentTreeItem(ti),
```

Comment:
Logic is backwards here.
```suggestion
            isLeaf: !isAzExtParentTreeItem(ti),
```